src/components/CohortBuilder/SQONProvider.js

```diff
@@ -112,17 +119,27 @@ class SQONProvider extends React.Component {
       setActiveSqonIndex,
       getActiveExecutableSqon: this.getActiveExecutableSqon,
       mergeSqonToActiveIndex: this.mergeSqonToActiveIndex,
+      sqonDictionary,
     });
   }
 }
 
-const mapStateToProps = state => {
+const mapStateToProps = (state) => {
   const { sqons, activeIndex, uid, virtualStudyId } = state.currentVirtualStudy;
+  const { sets } = state.saveSets.userSets;
+
+  let sqonDisctionary = [];
```

Comment:
On a personal level, I prefer a more functional style. For example: const sqonDictionnary = (sets || []).map(s => ({ setId: s.setId, tag: s.tag}))